web/ASC.Web.Client/src/components/pages/Home/index.js

```diff
@@ -2,80 +2,59 @@ import React, { useEffect } from "react";
 import PropTypes from "prop-types";
 import { withRouter } from "react-router";
 import { useTranslation } from "react-i18next";
-import styled from "styled-components";
 import Text from "@appserver/components/text";
 import toastr from "studio/toastr";
 import PageLayout from "@appserver/common/components/PageLayout";
-import history from "@appserver/common/history";
-import ModuleTile from "./ModuleTile";
 import { tryRedirectTo } from "@appserver/common/utils";
 import { setDocumentTitle } from "../../../helpers/utils";
 import { inject, observer } from "mobx-react";
-import config from "../../../../package.json";
-
-const HomeContainer = styled.div`
-  padding: 62px 15px 0 15px;
-  margin: 0 auto;
-  max-width: 1140px;
-  width: 100%;
-  box-sizing: border-box;
-  /*justify-content: center;*/
-
-  .home-modules {
-    display: flex;
-    flex-wrap: wrap;
-    margin: 0 -15px;
-
-    .home-module {
-      flex-basis: 0;
-      flex-grow: 1;
-      max-width: 100%;
-    }
-  }
-
-  .home-error-text {
-    margin-top: 23px;
-    padding: 0 30px;
-    @media (min-width: 768px) {
-      margin-left: 25%;
-      flex: 0 0 50%;
-      max-width: 50%;
-    }
-    @media (min-width: 576px) {
-      flex: 0 0 100%;
-      max-width: 100%;
-    }
-  }
-`;
-
-const Tiles = ({ modules, isPrimary }) => {
+import { HomeIllustration, ModuleTile, HomeContainer } from "./sub-components";
+import Heading from "@appserver/components/heading";
+
+const Tiles = ({ availableModules, username, t }) => {
   let index = 0;
-  const mapped = modules.filter(
-    (m) => m.isPrimary === isPrimary && m.isolateMode !== true
+  const { firstName, lastName } = username;
+  const fullName = `${firstName} ${lastName}`.trim();
+
+  const getGreeting = () => {
+    const time = new Date().getHours();
+
+    if (time >= 5 && time <= 11) return t("GoodMorning");
+    if (time >= 12 && time <= 16) return t("GoodAfternoon");
+    return t("GoodEvening");
+  };
+
+  const greetingMessage = `${getGreeting()}, ${fullName}!`;
```

Comment:
rewrite the function getGreeting  so that you immediately receive a ready greeting